habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -467,3 +467,45 @@ def step(self, ee_pos, **kwargs):
             self._sim.viz_ids["ee_target"] = self._sim.visualize_position(
                 global_pos, self._sim.viz_ids["ee_target"]
             )
+
+
+@registry.register_task_action
+class HumanoidJointAction(RobotAction):
+    def __init__(self, *args, sim: RearrangeSim, **kwargs):
+        super().__init__(*args, sim=sim, **kwargs)
+        self._sim: RearrangeSim = sim
+        self.num_joints = 19
+
+    def reset(self, *args, **kwargs):
+        super().reset()
+
+    @property
+    def action_space(self):
+        num_joints = self.num_joints
+        num_dim_transform = 16
+        # The action space is the number of joints plus 16 for a 4x4 transformtion matrix for the base
+        return spaces.Dict(
+            {
+                "human_joints_trans": spaces.Box(
+                    shape=(4 * (num_joints + num_dim_transform),),
+                    low=-1,
+                    high=1,
+                    dtype=np.float32,
+                )
+            }
+        )
+
+    def step(self, **kwargs):
+        new_pos_transform = kwargs["human_joints_trans"]
+        new_joints = new_pos_transform[:-16]
+        new_pos_transform = new_pos_transform[-16:]
+        if np.array(new_pos_transform).sum() != 0:
```

Comment:
why is this sum to 0 a good check? Can you add a comment here?

Reply:
This is a bit hacky... The high level policy calls all the actions at every step, even if those are not executed. When they are not executed, these are simply called as an array of 0s. This works fine in most action spaces since they are relative to the previous agent state, but here the action is absolute. I could make it relative, but given that it involves quaternions and a transformation matrix, it feels like it can complicate things.